packages/front/src/lib/resources/establishments/establishment.port.ts

```diff
@@ -1,21 +1,27 @@
-import type { DocumentDto } from "dto";
+import type { DocumentDto, Grant, Siret } from "dto";
 import requestsService from "$lib/services/requests.service";
 
 class EstablishmentPort {
-    incExtractData(identifier) {
-        const path = `/etablissement/${identifier}/extract-data`;
-        requestsService.get(path).catch(() => null);
+    getResource(siret: Siret, resource?: string) {
+        return requestsService.get(`/etablissement/${siret}${resource ? "/" + resource : ""}`);
     }
 
-    async getBySiret(identifier) {
-        const path = `/etablissement/${identifier}`;
-        return (await requestsService.get(path))?.data?.etablissement;
+    incExtractData(siret: Siret) {
+        this.getResource(siret, "extract-data").catch(() => null);
     }
 
-    async getDocuments(identifier): Promise<DocumentDto[]> {
-        const anwser = await requestsService.get(`/etablissement/${identifier}/documents`);
+    async getBySiret(siret: Siret) {
+        return (await this.getResource(siret))?.data?.etablissement;
+    }
+
+    async getDocuments(siret: Siret): Promise<DocumentDto[]> {
+        const anwser = await this.getResource(siret, "documents");
         return anwser?.data?.documents;
     }
+
+    async getGrants(siret: Siret): Promise<Grant[]> {
+        return (await this.getResource(siret, "grants"))?.data?.subventions;
```

Comment:
```suggestion
        return (await this.getResource(siret, "grants"))?.data?.subventions || [];
```

sinon tu as pas de valeur par défaut

Reply:
J'ai plutôt ajouté `as GetGrantsResponseDto` car côté API on renvoie forcément quelque chose et en fait ça utilise le DTO des deux côté donc je trouve ça cool

Reply:
C'est pas tout à fait le même sujet le type et la valeur par défaut. Si ya un pb et que ça renvoie nimp', c'est bien d'avoir une valeur par défaut même si ts est content avec le type

Reply:
oui mais à l'inverse avoir un type par défaut rend une erreur muette et passe inaperçu;

à la limite je préfère utiliser un if (subventions.length) throw error(). Je vais rajouter ça dans ma branche sur le nouveau tableau comme ça on en reparle à ce moment là si ça ne te convient pas (mais j'aimerai merge celle-ci)